engine/src/cubos/engine/input/bindings.cpp

```diff
@@ -0,0 +1,314 @@
+#include <sstream>
+
+#include <cubos/core/data/deserializer.hpp>
+#include <cubos/core/data/serializer.hpp>
+#include <cubos/core/log.hpp>
+
+#include <cubos/engine/input/bindings.hpp>
+
+using cubos::core::data::Deserializer;
+using cubos::core::data::Serializer;
+using cubos::core::io::Key;
+using cubos::core::io::keyToString;
+using cubos::core::io::Modifiers;
+using cubos::core::io::stringToKey;
+using namespace cubos::engine;
+
+const Key& KeyWithModifier::getKey() const
+{
+    return mKey;
+}
+
+const Modifiers& KeyWithModifier::getModifiers() const
+{
+    return mModifiers;
+}
+
+Key& KeyWithModifier::getKey()
+{
+    return mKey;
+}
+
+Modifiers& KeyWithModifier::getModifiers()
+{
+    return mModifiers;
+}
+
+std::string KeyWithModifier::toString() const
+{
+    std::stringstream ss;
+    ss << keyToString(mKey);
+
+    if ((mModifiers & Modifiers::Shift) != Modifiers::None)
+    {
+        ss << "+Shift";
+    }
+
+    if ((mModifiers & Modifiers::Control) != Modifiers::None)
+    {
+        ss << "+Control";
+    }
+
+    if ((mModifiers & Modifiers::Alt) != Modifiers::None)
+    {
+        ss << "+Alt";
+    }
+
+    if ((mModifiers & Modifiers::System) != Modifiers::None)
+    {
+        ss << "+System";
+    }
+
+    return ss.str();
+}
```

Comment:
This method is unnececessary since the type is serializable anyway! I would make both `io::Key` and `io::Modifiers` serializable, and then, when you need debug printing, just use the `Debug` utility to print serializable types.